packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-modal.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { ChangeEvent, KeyboardEvent } from 'react';
+import { css } from '@emotion/css';
+import {
+  spacing,
+  uiColors,
+  Button,
+  Modal,
+  H3,
+  TextInput,
+  FormFooter,
+  Select,
+  Option,
+  OptionGroup,
+} from '@mongodb-js/compass-components';
+
+import FormFieldContainer from '../../form-field-container';
+import { editableUrlOptions, UrlOption } from '../../../utils/url-options';
+
+const modalContainerStyles = css({
+  padding: 0,
+  'button[aria-label="Close modal"]': {
+    position: 'absolute',
+  },
+});
+
+const modalContentStyles = css({
+  padding: spacing[5],
+});
+
+const modalFooterStyles = css({
+  boxShadow: 'none',
+  border: 'none',
+  borderTop: `1px solid ${uiColors.gray.light2}`,
+});
+
+function UrlOptionsModal({
+  onClose,
+  selectedOption,
+  onUpdateOption,
+}: {
+  onClose: (isOpen: boolean) => void;
+  onUpdateOption: (option: UrlOption) => void;
+  selectedOption?: UrlOption;
+}): React.ReactElement {
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [option, setOption] = React.useState(
+    selectedOption ?? { key: '', value: '' }
+  );
+
+  const handleKeyPress = (event: KeyboardEvent<HTMLInputElement>): void => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      addUrlOption();
+    }
+  };
+
+  const addUrlOption = () => {
+    setErrorMessage('');
+    if (!option.key) {
+      return setErrorMessage('Please select the options key.');
+    }
+    onUpdateOption(option as UrlOption);
+  };
+
+  return (
+    <Modal
+      contentClassName={modalContainerStyles}
+      open={true}
+      setOpen={onClose}
+    >
+      <div className={modalContentStyles}>
+        <H3>Add custom url option</H3>
+        <FormFieldContainer>
+          <Select
+            label="Key"
+            placeholder="Select key"
+            name="key"
+            onChange={(key) => {
```

Comment:
For some reason this function is not called in my local setup. Am I doing anything wrong here?

Reply:
I think we have to set `usePortal={false}` when we're using `Select` in a modal.
Here's another place we use it:
https://github.com/mongodb-js/compass/blob/main/packages/databases-collections/src/components/collection-fields/time-series-fields.jsx#L90
We also style the z-index:
https://github.com/mongodb-js/compass/blob/main/packages/databases-collections/src/components/collection-fields/time-series-fields.module.less#L1